File: src/services/scanner/Scanner/Utils/ApiResult.cs
using Microsoft.AspNetCore.Mvc;

namespace Scanner;

public class ApiResult : JsonResult
{
    public ApiResult() : this(new ApiResultData())
    {
    }

    public ApiResult(List<string> messages) : this(new ApiResultData(messages))
    {
    }

    public ApiResult(object data) : this(new ApiResultData(data))
    {
    }

    public ApiResult(Exception exception) : this(new ApiResultData(new List<string> { exception.Message }))
    {
    }

    private ApiResult(ApiResultData data) : base(data)
    {
        ContentType = new System.Net.Mime.ContentType("application/json; charset=utf-8").ToString();
    }
}

File: src/clients/FU/data/src/main/java/com/example/scanner/data/storage/models/ApiResult.kt
package com.example.scanner.data.storage.models

class ApiResult<T> (
    val success: Boolean,
    val messages: List<String>,
    val data: List<T>?,
    var responseCode: Int
){

}

class ApiResultSingle<T> (
    val success: Boolean,
    val messages: List<String>,
    val data: T?,
    var responseCode: Int
){

}